plugins/GSdx/Renderers/DX11/GSOsdManagerDX11.cpp

```diff
@@ -0,0 +1,266 @@
+#include "stdafx.h"
+#include "GSOsdManagerDX11.h"
+
+GSOsdManagerDX11::GSOsdManagerDX11(ID3D11Device *dev)
+{
+	// NOTE: workaround for Fraps/Afterburner
+	if (!m_log_enabled && !m_monitor_enabled)
+		return;
+
+	HRESULT hr = E_FAIL;
+
+	D2D1_FACTORY_OPTIONS d2d_factory_options = {};
+
+	#ifdef DEBUG
+		d2d_factory_options.debugLevel = D2D1_DEBUG_LEVEL_INFORMATION;
+	#endif
+
+	CComPtr<IDXGIDevice> dxgid;
+	CComPtr<ID2D1Factory1> d2d_factory;
+
+	hr = D2D1CreateFactory(
+		D2D1_FACTORY_TYPE_SINGLE_THREADED,
+		__uuidof(ID2D1Factory1),
+		&d2d_factory_options,
+		(void **)&d2d_factory
+	);
+
+	if (FAILED(hr))
+	{
+		fprintf(stderr, "NOTE: DX11 OSD can only be enabled on Windows 7 with the Platform Update.\n");
+		fprintf(stderr, "Disabling OSD will remove this warning.\n");
+		fprintf(stderr, "https://support.microsoft.com/en-us/help/2670838/platform-update-for-windows-7-sp1-and-windows-server-2008-r2-sp1\n");
+		return;
+	}
+
+	hr = dev->QueryInterface(IID_PPV_ARGS(&dxgid));
+
+	hr = DWriteCreateFactory(
+		DWRITE_FACTORY_TYPE_SHARED,
+		__uuidof(IDWriteFactory),
+		reinterpret_cast<IUnknown **>(&m_dwrite_factory)
+	);
+
+	hr = d2d_factory->CreateDevice(dxgid, &m_dev_2d);
```

Comment:
`hr` is assigned but not used multiple times.

Reply:
I did an early return on failure for those.